plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorSmokeTest.java

```diff
@@ -303,6 +305,96 @@ public void testRegisterTableWithMetadataFile()
         assertUpdate(format("DROP TABLE %s", tableName));
     }
 
+    @Test
+    public void testUnregisterTable()
+    {
+        String tableName = "test_unregister_table_" + randomNameSuffix();
+
+        assertUpdate("CREATE TABLE " + tableName + " AS SELECT 1 a", 1);
+        String tableLocation = getTableLocation(tableName);
+
+        assertUpdate("CALL system.unregister_table(CURRENT_SCHEMA, '" + tableName + "')");
```

Comment:
```
        assertUpdate("CALL system.unregister_table(CURRENT_SCHEMA, '" + tableName + "$data')");
```

I tried this statement and it obviously failed.
```
Caused by: io.trino.spi.connector.TableNotFoundException: Table 'tpch.test_unregister_table_ul28evykib$data' not found
	at io.trino.plugin.iceberg.catalog.hms.TrinoHiveCatalog.lambda$unregisterTable$8(TrinoHiveCatalog.java:297)
```

I don't really think this has relevance, but I'm bringing it here just for information.